### PR TITLE
feat(tile): add experimental tiles

### LINKED ISSUE #1282: Experimental Component - Tile - Development
@laurenmrice commented on [Thu Oct 18 2018](https://github.com/carbon-design-system/carbon-website/issues/285)

##### Development
A developer can discover and install the experimental component by following accessible documentation.

- [ ] Add to carbon-components
- [ ] Update website (add to the experimental section and update component status page)

### PR DESCRIPTION
**Pull request by tw15egan**
Closes #1282 

Adds in Experimental Tile, and all variations (clickable, selectable, expandable, default)

#### Changelog

**New**

- Tile component and variations
- A new mixin for focus states `@focus-outline--x`

**Removed**

- Extra styles

#### Testing / Reviewing

Staging Link: http://carbon-dev-environment-exp-tile-intelligent-mongoose.mybluemix.net/?nav=tile

screen shot 2018-10-18 at 2 06 17 pm

### CODE REVIEW

File: src/globals/scss/_helper-mixins.scss [hunk 1]
```diff
@@ -56,6 +56,11 @@
   }
 }
 
+@mixin focus-outline--x {
+  outline: 2px solid $brand-01;
+  outline-offset: -2px;
+}
+
```
Comment (alisonjoseph) [inline src/globals/scss/_helper-mixins.scss:63]: Should we add this to this mixin and just re-name it to outline? I had added a `$focus` variable for the color (should maybe be `$focus-color`?) 

```
@mixin focus-outline($type: 'border') {
  @if ($type == 'border') {
    outline: 1px solid $brand-01;
  }

  @if ($type == 'blurred') {
    box-shadow: 0 0 0 3px $focus;
    outline: 1px solid transparent;
  }

  @if ($type == 'outline') {
    outline: 2px solid $focus;
    outline-offset: -2px;
  }
}
```

Or I'm totally fine w/ leaving it for now and just making sure we do a pass on all the components when we are done to try and consolidate how we are doing focus states. 😄

**Comment by laurenmrice** [comment]
looks good to me! 🙌

File: src/components/tile/_tile.scss [hunk 2]
```diff
@@ -170,3 +170,151 @@
...
+  .#{$prefix}--tile--is-selected {
+    outline: 1px solid $ui-04;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--tile--is-selected .bx--tile__checkmark {
+    opacity: 1;
+  }
+
+  .#{$prefix}--tile-input:checked {
```
Comment (joshblack) [inline src/components/tile/_tile.scss]: Could we do:

```scss
 .#{$prefix}--tile-input:checked + .#{$prefix}--tile__checkmark > svg {
  // ...
}
```

Just to make it a little easier to find the adjacent selector, and then to be more specific around `svg` styling.

**Review by joshblack — APPROVED** [review]
Looks great! Left a small note 👍

File: src/components/tile/_tile.scss (hunk 2 above)
Comment (tw15egan) [reply src/components/tile/_tile.scss]: Updated 😄

File: src/globals/scss/_helper-mixins.scss (hunk 1 above)
Comment (tw15egan) [reply src/globals/scss/_helper-mixins.scss:63]: Yeah I like this implementation, let me switch it to that

**Comment by tw15egan** [comment]
@alisonjoseph switched to your implementation, makes sense to extend the existing rather than create a new one 👍

**Review by alisonjoseph — APPROVED** [review]
LGTM! Thanks TJ! 🎉

**Pull request merged**